File: src/year2015/day02/part1.cc
#include <cstdint>
#include <string>

#include "../solutions.h"
#include "parse.h"

auto Year2015::Day02::Part1(const std::string& input) -> std::string {
  auto boxes = parse(input);
  uint32_t wrappingPaperRequired = 0;
  for (const auto& box : boxes) {
    wrappingPaperRequired +=
        (3 * box[0] * box[1] + 2 * box[2] * (box[0] + box[1]));
  }
  return std::to_string(wrappingPaperRequired);
}

File: src/year2015/day02/part1.rs
use super::parse::parse;
pub fn part1(input: String) -> Result<String, String> {
  let boxes = parse(input);
  if boxes.is_err() {
    return Err(boxes.unwrap_err());
  }
  let mut wrapping_paper_area = 0;
  for curr_box in boxes.unwrap() {
    wrapping_paper_area += 3 * curr_box[0] * curr_box[1] + 2 * curr_box[2] * (curr_box[0] + curr_box[1]);
  }
  return Ok(wrapping_paper_area.to_string());
}
